packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -144,12 +139,35 @@ async function proxyPreviewModeApiRouteHandler(
     .map(cookie => serialize(cookie.name, cookie.value, cookie as CookieSerializeOptions))
     .join(';')
 
+  const originalCookies = req.headers.cookie
```

Comment:
Since we've created the proxy headers here it might be easier to use the `Headers` APIs to get the cookie, set the new one, etc.?